cpp/sophus/image/proto/conv.h

```diff
@@ -9,11 +9,33 @@
 #pragma once
 
 #include "sophus/image.pb.h"
-#include "sophus/image/image_size.h"
+#include "sophus/image/dyn_image_types.h"
 
 namespace sophus {
 
 sophus::ImageSize fromProto(proto::ImageSize const& proto);
 proto::ImageSize toProto(sophus::ImageSize const& image_size);
 
+sophus::ImageLayout fromProto(proto::ImageLayout const& proto);
+proto::ImageLayout toProto(sophus::ImageLayout const& layout);
+
+Expected<sophus::PixelFormat> fromProto(proto::PixelFormat const& proto);
+proto::PixelFormat toProto(sophus::PixelFormat const& layout);
+
+Expected<sophus::AnyImage<>> fromProto(proto::DynImage const& proto);
+Expected<sophus::IntensityImage<>> intensityImageFromProto(
+    proto::DynImage const& proto);
+
+template <class TPredicate>
+proto::DynImage toProto(sophus::DynImage<TPredicate> const& image) {
```

Comment:
why not take an `AnyImage`?

Reply:
This is more general. It takes AnyImages, IntensityImages, ... AnyDynImage no matter what TPredicate.